src/constants/clickTrackingAttribute.js

```diff
@@ -0,0 +1 @@
+export default "data-alloy-key";
```

Comment:
Do you think this will be used for components other than personalization? If not, then this file should probably be stored inside the personalization directory. If it will be used for components other than personalization, then I think how `data-alloy-key` values are assigned and how storage is managed will need some tweaking (either now or later). Just thinking out loud a bit here, but right now the UUID assigned to the elements are specific to the "click" `settings` object. In other words, multiple elements will share the same UUID. This works for the personalization case, but doesn't work for the generic case of wanting to store some information for a specific element. You may have thought about this already and are thinking we resolve it at the point in time that we need to share the storage across components. Anyway, let me know if you want to talk it through.

Reply:
I have moved this to `constants` assuming that data collector click tracking functionality would need to know the name of the attribute when it starts to scan the DOM elements. Once the `data-alloy-key` attribute is found, it's value will be passed into `click` life-cycle hook and personalization will have to subscribe to this hook, similar to `onResponse`.

Probably something will change when @liljenback will finish his work, but I don't think we will have a lot of changes.

Reply:
I thought we had discussed having DataCollector either (1) pass the element that was clicked (instead of searching for elements with `data-alloy-key`) or (2) pass an array of elements, where each one is an element in the hierarchy of the element that was clicked. In each case, Personalization would be the only one aware of `data-alloy-key`.  I think it makes more sense to get rid of `data-alloy-key` and instead have DataCollector pass the element that was clicked and then personalization would iterate over the hierarchy and look up each element in the WeakMap to see if there's an entry (e.g., `metaByElement.has(element)`)

Reply:
@Aaronius is correct. This way other Components in the future can implement this lifecycle hook without it being specific to Personalization.